diff -Nrup fsam7400-0.5.1.orig/fsam7400.c fsam7400-0.5.1/fsam7400.c
--- fsam7400-0.5.1.orig/fsam7400.c	2006-09-24 23:11:08.000000000 +1000
+++ fsam7400-0.5.1/fsam7400.c	2006-10-16 20:41:16.000000000 +1000
@@ -28,7 +28,9 @@
 
 *******************************************************************************/
 
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #ifdef CONFIG_X86
 #include <linux/version.h>
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)
