diff -Nrup fsam7400-0.5.1.orig/fsam7400.c fsam7400-0.5.1/fsam7400.c
--- fsam7400-0.5.1.orig/fsam7400.c	2006-09-24 23:11:08.000000000 +1000
+++ fsam7400-0.5.1/fsam7400.c	2006-10-16 20:46:27.000000000 +1000
@@ -45,7 +45,11 @@
 #include <linux/ctype.h>
 #include <linux/string.h>
 #include <linux/kmod.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,19)
+#include <linux/io.h>
+#else
 #include <asm/io.h>
+#endif
 #include <asm/uaccess.h>
 
 #define DRV_NAME         "fsam7400"
