add kernel 2.6.24 compatibility.
submitted upstream 2007-12-16.

--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@ KERNELMAJOR=$(shell echo $(KERNELVERSION
 CONFIG_FSAM7400?=m
 obj-$(CONFIG_FSAM7400) += fsam7400.o
 
-CFLAGS+=-c -Wall -Wstrict-prototypes -Wno-trigraphs -O2 -fomit-frame-pointer -fno-strict-aliasing -fno-common -pipe
+EXTRA_CFLAGS+=-c -Wall -Wstrict-prototypes -Wno-trigraphs -O2 -fomit-frame-pointer -fno-strict-aliasing -fno-common -pipe
 INCLUDE=-I$(KERNELSRC)/include
 
 ifeq ($(KERNELMAJOR), 2.6)
@@ -29,7 +29,7 @@ fsam7400.ko: $(SOURCE) 
 	$(MAKE) -C $(KERNELSRC) SUBDIRS=$(PWD) modules
 
 fsam7400.o: $(SOURCE)
-	$(CC) $(INCLUDE) $(CFLAGS) -DDEBUG -DMODVERSIONS -DMODULE -D__KERNEL__ -o $(TARGET) $(SOURCE)
+	$(CC) $(INCLUDE) $(EXTRA_CFLAGS) -DDEBUG -DMODVERSIONS -DMODULE -D__KERNEL__ -o $(TARGET) $(SOURCE)
 
 clean:
 	@echo -n "sweeping directory... "
